cmd/tools/rest/HrefBuilder.go

```diff
@@ -0,0 +1,79 @@
+package rest
+
+import "strings"
+
+type HrefBuilder struct {
+	apiPath       string
+	fields        string
+	filter        []string
+	queryFields   string
+	queryValue    string
+	maxRecords    string
+	returnTimeout string
+}
+
+func NewHrefBuilder() *HrefBuilder {
+	return &HrefBuilder{}
+}
+
+func (b *HrefBuilder) APIPath(apiPath string) *HrefBuilder {
+	b.apiPath = apiPath
+	return b
+}
+
+func (b *HrefBuilder) Fields(fields string) *HrefBuilder {
```

Comment:
instead of requiring all the consumers to do the join dance, might be better to change this to take a slice arg and do the join in build when the slice is not empty